src/components/LmsConfigurations/DegreedIntegrationConfigForm.test.jsx

```diff
@@ -0,0 +1,113 @@
+describe('<DegreedIntegrationConfigForm />', () => {
+  test('renders Degreed Config Form', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    // Verify all expected fields are present.
+    screen.getByLabelText('Active');
+    screen.getByLabelText('Degreed User ID');
+    screen.getByLabelText('Degreed User Password');
+    screen.getByLabelText('Degreed Instance URL');
+    screen.getByLabelText('Degreed Organization Code');
+    screen.getByLabelText('API Client ID');
+    screen.getByLabelText('API Client Secret');
+  });
+
+  test('required fields show as invalid when not filled in', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    fireEvent.click(screen.getByText('Submit'));
+
+    expect(screen.getByLabelText('Degreed User ID')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed User Password')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed Instance URL')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed Organization Code')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('API Client ID')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('API Client Secret')).toHaveClass('is-invalid');
+    expect(screen.getByRole('alert')).toHaveTextContent(
+      'Form could not be submitted as there are fields with invalid values. '
+      + 'Please correct them below.',
+    );
+  });
+
+  test('submit calls postNewDegreedConfig when config is not present', () => {
+    const expectedFormData = {
+      degreed_user_id: 'testuserid',
+      degreed_user_password: 'testpass',
+      degreed_base_url: 'testinstance',
+      degreed_company_id: 'testorg',
+      key: 'testclientid',
+      secret: 'testsecret',
+      enterprise_customer: enterpriseId,
+    };
+
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    fireEvent.change(screen.getByLabelText('Degreed User ID'), {
+      target: { value: 'testuserid' },
+    });
+    fireEvent.change(screen.getByLabelText('Degreed User Password'), {
+      target: { value: 'testpass' },
+    });
+    fireEvent.change(screen.getByLabelText('Degreed Instance URL'), {
+      target: { value: 'testinstance' },
+    });
+    fireEvent.change(screen.getByLabelText('Degreed Organization Code'), {
+      target: { value: 'testorg' },
+    });
+    fireEvent.change(screen.getByLabelText('API Client ID'), {
+      target: { value: 'testclientid' },
+    });
+    fireEvent.change(screen.getByLabelText('API Client Secret'), {
+      target: { value: 'testsecret' },
+    });
+    fireEvent.click(screen.getByText('Submit'));
+
+    const postedFormData = Array.from(LmsApiService.postNewDegreedConfig.mock.calls[0][0].entries())
+      .reduce((acc, f) => ({ ...acc, [f[0]]: f[1] }), {});
+
+    expect(postedFormData).toMatchObject(expectedFormData);
+  });
+
+  test('submit calls updateDegreedConfig when config is not present', () => {
```

Comment:
The only other test I would like is one that verifies that the initial config is being displayed to the user. So, rendering with an initialConfig and checking the text of `screen.getByLabelText('Active');` to make sure it's showing the initial config.

Reply:
Ah gotcha. Yeah I can add that.